### PR TITLE
enh(meta): use select2 for metric selection in META form

### PR DESCRIPTION
**Pull request by adr-mo**
This PR intends to replace the simple select used in the META service form by a select2.
This change is mandatory for servers that have a large amount of metrics. When opening. the select, the page stopped working.

**Fixes** # (issue)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.10.x
- [ ] 20.04.x
- [x] 20.10.x
- [x] 21.04.x (master)

<h2> How this pull request can be tested ? </h2>

- Make sure to have a large amount of metrics
- Add a META and select a METRIC in the META form
- 
## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/api/class/centreon_metric.class.php [hunk 1]
```diff
@@ -111,9 +111,62 @@ public function getList()
                 'text' => $row['metric_name']
             );
         }
+
         return $metrics;
     }
 
+        /**
```
Comment (kduret) [inline www/api/class/centreon_metric.class.php:118]:
```suggestion
    /**
```

File: www/include/configuration/configObject/meta_service/formMetaService.php [hunk 2]
```diff
@@ -126,6 +117,12 @@
     'multiple' => false,
     'linkedObject' => 'centreonTimeperiod'
 );
+$metricAvRoute = './include/common/webServices/rest/internal.php?object=centreon_metric&action=listMetrics';
```
Comment (kduret) [inline www/include/configuration/configObject/meta_service/formMetaService.php:120]:
```suggestion
$metricAvRoute = './api/internal.php?object=centreon_metric&action=listMetrics';
```

File: www/api/class/centreon_metric.class.php [hunk 3]
```diff
@@ -111,9 +111,62 @@ public function getList()
...
+                $acl->getServicesString('ID', $this->pearDBMonitoring) . ') ';
+        }
+
+        $query .= ' ORDER BY `metric_name` COLLATE utf8_general_ci ';
+        $stmt = $this->pearDBMonitoring->prepare($query);
+        $stmt->bindParam(':name', $queryValues['name'], \PDO::PARAM_STR);
+        $dbResult = $stmt->execute();
+        if (!$dbResult) {
+            throw new \Exception("An error occured");
+        }
```
Comment (kduret) [inline www/api/class/centreon_metric.class.php:155]: PDO is already configured to throw an exception if an error occured
```suggestion
```

**Comment by loiclau** [comment]
cant use getList() and make a count on this?

**Comment by kduret** [comment]
> cant use getList() and make a count on this?

No because we need to get `total`

**Review by callapa — APPROVED** [review]

**Review by loiclau — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Pull request merged**